common/src/main/java/com/wynntils/features/combat/AutoAttackFeature.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.combat;
+
+import com.wynntils.core.components.Models;
+import com.wynntils.core.consumers.features.Feature;
+import com.wynntils.core.persisted.config.Category;
+import com.wynntils.core.persisted.config.ConfigCategory;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.ArmSwingEvent;
+import com.wynntils.mc.event.ArmSwingEvent.ArmSwingContext;
+import com.wynntils.mc.event.ChangeCarriedItemEvent;
+import com.wynntils.mc.event.SetSlotEvent;
+import com.wynntils.mc.event.TickEvent;
+import com.wynntils.mc.event.UseItemEvent;
+import com.wynntils.models.character.type.ClassType;
+import com.wynntils.models.spells.event.SpellEvent;
+import com.wynntils.models.spells.type.SpellDirection;
+import com.wynntils.utils.mc.LoreUtils;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.wynn.ItemUtils;
+import net.minecraft.client.player.LocalPlayer;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.bus.api.SubscribeEvent;
+
+@ConfigCategory(Category.COMBAT)
+public class AutoAttackFeature extends Feature {
+    private static final int TICKS_PER_ATTACK = 2;
+
+    private Boolean canUseWeapon = null;
```

Comment:
Don't use `Boolean` with `null` as a tri-state. Either make an nested enum, or two different booleans, one which says if this is checked, and one what the value is.

Reply:
Not sure on the enum member names, but probably good enough (also went with NOT_USABLE instead of UNUSABLE for a little bit better readability at a quick glace).